kotlin/samples/todo-android/common/src/main/java/com/squareup/sample/todo/MasterDetailAware.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.sample.todo
+
+/**
+ * Informs [MasterDetailAware] renderings whether they're children of a [MasterDetailScreen],
+ * and if so in what configuration.
+ */
+enum class MasterDetailConfig {
+  /**
+   * Parent is not a [MasterDetailScreen]. Expected to be used as the default value
+   * of [MasterDetailAware.masterDetailConfig].
+   */
+  None,
+
+  /**
+   * Rendering is in [MasterDetailScreen.masterRendering], running in split screen
+   * configuration.
+   */
+  Master,
+
+  /**
+   * Rendering is in [MasterDetailScreen.detailRendering], running in split screen
+   * configuration.
+   */
+  Detail,
+
+  /**
+   * Rendering is the only visible child in a [MasterDetailScreen] running in
```

Comment:
Seems odd that single pane children can't know if they're master or detail. OTOH I haven't actually needed that info.